File: backend/src/main/java/ch/rafaelurben/sheetmusiclearner/backend/io/async/dto/request/PieceSectionAddRequestDto.java
/* (C) 2026 - Rafael Urben */
package ch.rafaelurben.sheetmusiclearner.backend.io.async.dto.request;

import java.util.UUID;

public record PieceSectionAddRequestDto(
    int position,
    int timeSignatureNumerator,
    int timeSignatureDenominator,
    int barCount,
    int bpm,
    UUID scoreSheetId,
    Float posX1,
    Float posY1,
    Float posX2,
    Float posY2) {}

File: frontend/src/interfaces/async/request/piece/PieceSectionAddRequestDto.ts
/*
 * (C) 2026. - Rafael Urben
 */

export default interface PieceSectionAddRequestDto {
  position: number;
  name: string;
  timeSignatureNumerator: number;
  timeSignatureDenominator: number;
  barCount: number;
  bpm: number;
  scoreSheetId: string | null;
  posX1: number | null;
  posY1: number | null;
  posX2: number | null;
  posY2: number | null;
}
